core/aws_ddk_core/cicd/actions.py

```diff
@@ -110,3 +110,96 @@ def get_synth_action(
         commands=["cdk synth"],
         role_policy_statements=role_policy_statements,
     )
+
+
+def get_cfn_nag_action(
+    file_set_producer: Optional[IFileSetProducer], stage_name: Optional[str] = "CFNNag"
+) -> ShellStep:
+    """
+    Get CFN Nag action.
+
+    Parameters
+    ----------
+    file_set_producer: Optional[IFileSetProducer]
+        File set to run security scan on
+    stage_name: Optional[str]
+        Name for stage. Default is "CFNNag"
+
+    Returns
+    -------
+    action : ShellStep
+        Codebuild step
+    """
+    return ShellStep(
+        stage_name,
+        input=file_set_producer,
+        install_commands=["gem install cfn-nag"],
+        commands=[
+            "fnames=$(find ./ -type f -name '*.template.json')",
+            "for f in $fnames; do cfn_nag_scan --input-path $f; done",
+        ],
+    )
+
+
+def get_bandit_action(code_pipeline_source: CodePipelineSource, stage_name: Optional[str] = "Bandit") -> ShellStep:
+    """
+    Get Bandit action.
+
+    Parameters
+    ----------
+    code_pipeline_source: CodePipelineSource
+        Code Pipeline source stage
+    stage_name: Optional[str]
+        Name for stage. Default is "Bandit"
+
+    Returns
+    -------
+    action : CodeBuildStep
+        Synth action
+    """
+    return ShellStep(
+        stage_name,
+        input=code_pipeline_source,
+        install_commands=["pip install bandit"],
+        commands=["bandit -r -ll -ii ."],
+    )
+
+
+def get_tests_action(
+    file_set_producer: Optional[IFileSetProducer],
+    commands: Optional[List[str]] = None,
+    stage_name: Optional[str] = "Tests",
+) -> ShellStep:
+    """
+    Return shell script action that runs tests.
+
+    Parameters
+    ----------
+    scope : Construct
+        Scope within which this construct is defined
+    file_set_producer: Optional[IFileSetProducer]
+        File set to run tests on
+    commands: Optional[List[str]]
+        Additional commands to run in the test. Defaults to "./test.sh" otherwise
+    stage_name: Optional[str]
+        Name for stage. Default is "Bandit"
+
+    Returns
+    -------
+    action : ShellStep
+        Test action
+    """
+    install_commands: List[str] = []
+    commands = ["./test.sh || echo 'no test script'"] if commands is None else commands
```

Comment:
Does this suffice as our default behavior? Or is there some recursive `pytest ...` command we can run to run all possible tests?

Reply:
If we added test stage but there is no test script, I think we should let the pipeline fail. Also let's add test.sh into our init template?

Reply:
I like that, updating..